### PR TITLE
[CORL-159, CORL-160] Stream Config Tab

### PR DESCRIPTION
**Pull request by alinhle**
## What does this PR do?
- Implements the Stream Config Tab
- Upgrades `found`, and  `React`.
- Introduce `expectAndFail` that will immediately fail when expectation is not met. Useful when running inside a try-catch block.

## How do I test this PR?

- Login as moderator or admin
- Do your thing

### CODE REVIEW

File: src/core/client/framework/lib/form/SubmitHookHandler.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  private addSubmitHook: AddSubmitHook = hook => {
+    this.submitHooks.push(hook);
+    return () => {
+      this.submitHooks = this.submitHooks.filter(h => h !== hook);
+    };
+  };
+
+  public render() {
+    return (
+      <>
```
Comment (wyattjoh) [inline src/core/client/framework/lib/form/SubmitHookHandler.tsx:74]: What does the extra `<>` add here?

File: src/core/client/framework/lib/form/helpers.tsx [hunk 2]
```diff
@@ -44,6 +44,7 @@ export const formatPercentage = (v: any) => {
   return Math.round(v * 100).toString();
 };
 
+export const parseBool = (v: any) => !!v;
```
Comment (wyattjoh) [inline src/core/client/framework/lib/form/helpers.tsx]: Strong suggestion to replace with:

```suggestion
export const parseBool = (v: any) => Boolean(v);
```

File: src/core/server/models/story/index.ts [hunk 3]
```diff
@@ -61,6 +64,16 @@ export interface Story extends TenantResource {
...
+
+  /**
+   * moderation determines whether or not this is a PRE or POST moderated story.
+   */
+  moderation?: GQLMODERATION_MODE;
+
+  /**
+   * premodLinksEnable will put all comments that contain links into premod.
+   */
+  premodLinksEnable?: boolean;
```
Comment (wyattjoh) [inline src/core/server/models/story/index.ts:83]: These settings should be available inside `settings.moderation` on the Story.

File: src/core/client/framework/lib/form/SubmitHookHandler.tsx (hunk 1 above)
Comment (alinhle) [reply src/core/client/framework/lib/form/SubmitHookHandler.tsx:74]: Nothing :-)

File: src/core/client/framework/lib/form/helpers.tsx (hunk 2 above)
Comment (alinhle) [reply src/core/client/framework/lib/form/helpers.tsx]: image
Ok 😃

File: src/core/server/models/story/index.ts (hunk 3 above)
Comment (alinhle) [reply src/core/server/models/story/index.ts:83]: I see what you mean!

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**